src/combinatorics/derangements/finite.lean

```diff
@@ -0,0 +1,117 @@
+/-
+Copyright (c) 2021 Henry Swanson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Henry Swanson
+-/
+import combinatorics.derangements.basic
+import data.fintype.card
+import tactic.delta_instance
+import tactic.ring
+
+/-!
+# Derangements on fintypes
+
+This file contains lemmas that describe the cardinality of `derangements α` when `α` is a fintype.
+
+# Main definitions
+
+  - `card_derangements_invariant`: A lemma stating that the number of derangements on a type `α`
+    depends only on the cardinality of `α`.
+
+  - `num_derangements n`: The number of derangements on an n-element set, defined in a computation-
+    friendly way.
+
+  - `card_derangements_eq_num_derangements`: Proof that `num_derangements` really does compute the
+    number of derangements.
+
+  - `num_derangements_sum`: A lemma giving an expression for `num_derangements n` in terms of
+    factorials.
```

Comment:
For docgen
```suggestion
* `card_derangements_invariant`: A lemma stating that the number of derangements on a type `α`
    depends only on the cardinality of `α`.
* `num_derangements n`: The number of derangements on an n-element set, defined in a computation-
    friendly way.
* `card_derangements_eq_num_derangements`: Proof that `num_derangements` really does compute the
    number of derangements.
* `num_derangements_sum`: A lemma giving an expression for `num_derangements n` in terms of
    factorials.
```

Reply:
Does docgen not like `-`-as-bullet-points? Also, how can I check the output of docgen myself?